File: packages/frontend/src/lib/api/studyBlocks.ts
import { invoke } from "@tauri-apps/api/core";
import type { StudyBlock } from "./types";

export function createStudyBlock(
  spaceId: string,
  title: string,
  examId: string,
  date: string,
  startTime: string,
  endTime: string,
): Promise<StudyBlock> {
  return invoke("create_study_block", { spaceId, title, examId, date, startTime, endTime });
}

export function listStudyBlocks(spaceId: string): Promise<StudyBlock[]> {
  return invoke("list_study_blocks", { spaceId });
}

File: apps/desktop/src-tauri/src/commands/study_blocks.rs
use crate::db::study_blocks::{self, StudyBlock};
use crate::db::DbState;
use crate::error::AppResult;
use tauri::State;

#[tauri::command]
pub fn create_study_block(
    state: State<DbState>,
    space_id: String,
    title: String,
    exam_id: String,
    date: String,
    start_time: String,
    end_time: String,
) -> AppResult<StudyBlock> {
    let conn = state.0.lock().unwrap();
    study_blocks::create_study_block(&conn, space_id, title, exam_id, date, start_time, end_time)
}

#[tauri::command]
pub fn list_study_blocks(state: State<DbState>, space_id: String) -> AppResult<Vec<StudyBlock>> {
    let conn = state.0.lock().unwrap();
    study_blocks::list_study_blocks(&conn, &space_id)
}
